### DISCUSSION TITLE
Linking pyoptsparse to cyipopt

### DISCUSSION THREAD

**Discussion by pollinico** [mdolab/pyoptsparse#373] [Ideas]
Hi, I just installed PyOptSparse with SNOPT, and I am very excited about how well everything works! Thank you, guys!
Just a thought: Have you considered linking the installation of PyOptSparse to [cyipopt](https://github.com/mechmotum/cyipopt)? 
In this way, it would be easier to include IPOPT in the PyOptSparse environment. 
I don't know what this would entail; it goes beyond my current capabilities, but it would make everything much easier, I think. 
Please let me know what you think. 
Thank you for your attention, Nicolo.

**Comment by ewu63** [comment]
Yep, this is something I had thought of before. It is probably a better idea to use an existing Python wrapper for IPOPT rather than have our own wrapper, and those are likely packaged/available on PyPI/conda already. I have created an issue #374 to track this feature request. With most open-source packages, there are no promises that we will do this, but feel free to submit PRs if you want to tackle this yourself.
